app/controllers/result_filters/location_controller.rb

```diff
@@ -45,6 +42,10 @@ def build_results_filter_query_parameters
         .query_parameters_with_defaults
     end
 
+    def location_option_selected?
```

Comment:
I wonder if it makes sense to add `location_option_selected` and `across_england_selected` methods to the `LocationForm`.

Reply:
I'm not familiar enough to have an opinion on it. What does everyone else think about moving `location_option_selected?` and `across_england_selected?` methods to `LocationForm`

Reply:
just throwing this into the mist....

3 of those `option_selected?` looks like it should be a switch case statement....

```ruby
    def create
      # if searching for specific provider go to results page
      if filter_params[:l] == "3"
        redirect_to(provider_path(params_for_provider_search))
        return
      end

      form_params = strip(filter_params.clone)
      form_object = LocationFilterForm.new(form_params)
      if form_object.valid?
        all_params = form_params.merge!(form_object.params)

        if filter_params[:l] == "1"
          submitted_params = params_for_location_search(all_params)
        elsif filter_params[:l] == "2"
          submitted_params = params_for_across_england_search(all_params)
        end

        redirect_to(next_step(submitted_params))
      else
        flash[:error] = form_object.errors
        back_to_current_page_if_error(form_params)
      end
    end
```

Reply:
I think its easier to read if they are private methods but I have taken on board your comments and have pushed up a new commit. It reduces the number of private methods and should make the `create` method easier to read.

Reply:
😭  not sure why the tests are now failing on azure but passing locally